index.js

```diff
@@ -25,16 +29,172 @@ const countriesList = document.getElementById('countriesList');
 const submit = document.getElementById('submit');
 const output = document.getElementById('output');
 
-(async () => {
-    fromCountry.disabled = true;
-    toCountry.disabled = true;
-    submit.disabled = true;
+async function loadBordersCountryData(country) {
+    const countries = await getData(`https://restcountries.com/v3.1/alpha/${country}?fields=cca3&fields=borders`);
+    return countries.borders.reduce((result, neighbor) => {
+        result.push(neighbor);
+        return result;
+    }, []);
+}
+
+// path optimization
+function minimazePath(maxLength, allPath, uniqueElement) {
+    const resObject = {};
+    for (let i = 0; i < uniqueElement.length; i++) {
+        const currentCountry = uniqueElement[i];
+        let minValue = maxLength;
+        resObject[currentCountry] = { ph: [], dp: maxLength };
+        for (let j = 0; j < allPath.length; j++) {
+            const arr = allPath[j].split('>');
+            const index = arr.indexOf(currentCountry);
+            if (index >= 0 && index < minValue) {
+                minValue = index + 1;
+                if (minValue < resObject[currentCountry].dp) {
+                    resObject[currentCountry].ph = [];
+                    resObject[currentCountry].dp = minValue;
+                }
+                const element = arr.slice(0, minValue).join('>');
+                if (resObject[currentCountry].ph.includes(element) === false) {
+                    resObject[currentCountry].ph.push(element);
+                }
+            }
+        }
+    }
+    return resObject;
+}
+
+async function getAllPathCountry(fromCountry, countryNeighborCach, maxDeep) {
+    let countRequest = 0;
+    const uniqueElement = [];
+    // Find all possible paths from the given country
+    async function buildPath(fromCountry, deep) {
+        if (deep >= maxDeep) {
+            return [fromCountry];
+        }
+        let arrNeighbor;
+        // Let's see where to get information about the country server or cache
+        let flagCach = false;
+        if (countryNeighborCach.hasOwnProperty(fromCountry) === true) {
+            arrNeighbor = countryNeighborCach[fromCountry];
+            flagCach = true;
+        } else {
+            arrNeighbor = await loadBordersCountryData(fromCountry);
+            countRequest += 1;
+            countryNeighborCach[fromCountry] = [];
+        }
+        if (flagCach === false) {
+            const index = arrNeighbor.indexOf(fromCountry);
+            if (index > -1) {
+                arrNeighbor.splice(index, 1);
+            }
+            countryNeighborCach[fromCountry] = arrNeighbor.concat([]);
+        }
+        const resultArray = [];
+        for (let i = 0; i < arrNeighbor.length; i++) {
+            resultArray.push(buildPath(arrNeighbor[i], deep + 1));
+            if (uniqueElement.includes(arrNeighbor[i]) === false) {
```

Comment:
```suggestion
            if (!uniqueElement.includes(arrNeighbor[i])) {
```
Хотя что-то в этом есть. Глаз цепляется за `false`, и понятна суть проверки.